### ISSUE TITLE
`Query::column()` does not respect `Query::indexBy()`.

### ISSUE THREAD

**Issue by klimov-paul** [yiisoft/yii2-sphinx#65]
Extracted from https://github.com/yiisoft/yii2/issues/12649

`Query::column()` does not respect `Query::indexBy()`.

Unit test to reproduce:

```php
$result = (new Query)
            ->select('category_id')
            ->from('yii2_test_item_index')
            ->orderBy(['id' => SORT_DESC])
            ->indexBy('id')
            ->column($db);
        $this->assertEquals([2 => 2, 1 => 1], $result);
```

**Comment by klimov-paul** [comment]
Since `yii\sphinx\Query` extends `yii\db\Query` fix https://github.com/yiisoft/yii2/pull/12653 resolves this issue as well.

Unit test added 37cba64befd2736383e87f6433ec52226eb36794

**Comment by klimov-paul** [comment]
Build will fix after yii 2.0.10 release.
